### PR TITLE
Update UG for fee and to not allow for session fee to be 1000000 or more

### LINKED ISSUE #207: [PE-D] Fee page does not contain other months. No instructions on how i can create months
No details provided.

-------------
Labels: `severity.High` `type.FeatureFlaw`
original: Winniehyx/ped#13

### LINKED ISSUE #219: [PE-D] Precision error in `add_session`
Command input : 
`add_session n/Alex Yeoh d/2021-01-01 t/18:00 k/120 s/Biology f/999999999999999999999999999999999`

Expected : `fee` is `999999999999999999999999999999999`
Result : 

-------------
Labels: `severity.Low` `type.FunctionalityBug`
original: jessen11/ped#6

### LINKED ISSUE #220: [PE-D] Command `add_session` not working
input : `add_session n/Alex Yeoh d/2021-01-01 t/18:00 k/120 s/Biology f/0.0000000000000000000000000000000000000000000000000000000000000001`

Code rejects input, although every variable is valid. 

-------------
Labels: `severity.Medium` `type.FunctionalityBug`
original: jessen11/ped#7

### LINKED ISSUE #221: [PE-D] Fee of `infinity` appears
Input : `add_rec_session n/Alex Yeoh d/2021-01-01 e/2021-01-29 b/7 t/18:00 k/120 s/Biology f/99999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999`

Expected : fee equal to `99999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999`

Output: `fee` is infinity

-------------
Labels: `severity.Low` `type.FunctionalityBug`
original: jessen11/ped#8

### LINKED ISSUE #222: [PE-D] Infinity appear in `fee` function
Continuing in the last bug, 
command = `fee n/Alex Yeoh m/1 y/2021`
resulting in this 

-------------
Labels: `severity.Low` `type.FunctionalityBug`
original: jessen11/ped#9

### LINKED ISSUE #249: [PE-D] Issues with fee
Fee-only shows 3 months and it's the previous 3 months. 

Also, they allow taking in fees that are over the total GDP of the entire world. This does not make sense, there should be a cap on the tuition fee. 

-------------
Labels: `severity.Medium` `type.FeatureFlaw`
original: ong6/ped#21

### LINKED ISSUE #250: [PE-D] Wrong Fee
Delete a session that shows the fee as 1.01010101010101E42, seems to be unintentionally done

-------------
Labels: `severity.VeryLow` `type.FeatureFlaw`
original: ong6/ped#22

### PR DESCRIPTION
**Pull request by enhao25**
Resolves #220, Resolves #250, Resolves #222, Resolves #221, Resolves #219, Resolves #207 and Resolves #249.

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -122,8 +122,9 @@ The Home page is split into two views: **Reminder** and **Monthly Fees**.
 ##### Reminder
 The Reminder section displays tuition sessions within 3 days of the current date. Namely today, tomorrow and the day after tomorrow.
 
-##### Monthly Fees
-The Monthly Fees section displays tuition fees for the current and the past two months.
+##### 3 Months Monthly Fee
+The 3 Months Monthly Fees section displays tuition fees for the current month and the past two months.
+Note: The current month and past two months are calculated when the user first opened the application. No other months can be added to this section.
```
Comment (samleewy) [inline docs/UserGuide.md]: Some english: when the user first opens** the application

File: docs/UserGuide.md [hunk 2]
```diff
@@ -330,12 +331,20 @@ Adds a single tuition session to TutorBuddy.
 Format: `add_session n/STUDENT_NAME d/DATE t/TIME k/DURATION s/SUBJECT f/FEE`
 
 * `STUDENT_NAME` should match the exact student’s name in TutorBuddy
+* `DATE` should be the date of the session in `yyyy-mm-dd` format
 * `DURATION` should be in minutes
-* `FEE` should be the total tuition fee for the total duration
+* `FEE` should be the total tuition fee for 1 session of the given duration
+    * TutorBuddy only accepts `FEE` within an acceptable range of values between 0 (Voluntary Session) to below 10000 as we do not expect a single session to cost $10,000 or more.
```
Comment (samleewy) [inline docs/UserGuide.md]: Not sure if we should rename voluntary session to **free** instead? While reading it, thought it was a new kind of session HAHA

Also, not sure if we should justify why we chose 0 - $10,000, feels like we're doing it just not to get penalised HAHA maybe specifying the range is enough. either way works!

Comment (samleewy) [reply docs/UserGuide.md]: Actually on second thought after reading your codes below: maybe can rephrase it to, accepts `FEE` within an acceptable range of values between 0 to 9999.99 (both inclusive)

File: src/test/java/seedu/address/model/session/FeeTest.java [hunk 3]
```diff
@@ -32,14 +32,19 @@ public void isValidFee() {
         assertFalse(Fee.isValidFee("10.230"));
         assertFalse(Fee.isValidFee("005"));
         assertFalse(Fee.isValidFee("005.20"));
+        assertFalse(Fee.isValidFee("0.00000001"));
+        assertFalse(Fee.isValidFee("10000"));
+        assertFalse(Fee.isValidFee("1000000000000000"));
+        assertFalse(Fee.isValidFee("-1000000000000000"));
+        assertFalse(Fee.isValidFee("-0"));
+        assertFalse(Fee.isValidFee("-100.40"));
```
Comment (samleewy) [inline src/test/java/seedu/address/model/session/FeeTest.java]: Probably we can do a 9999.999 check as well for 2dp?

**Review by samleewy — COMMENTED** [review]
Some minor comments!

File: docs/UserGuide.md (hunk 1 above)
Comment (enhao25) [reply docs/UserGuide.md]: Nice! Updated in the latest commit!

File: docs/UserGuide.md (hunk 2 above)
Comment (enhao25) [reply docs/UserGuide.md]: Updated to "TutorBuddy only accepts `FEE` within an acceptable range of values between 0 and 999999.99 (both inclusive) as we do not expect a single session to cost more than $999,999.99 or more." as per suggestion.

File: src/test/java/seedu/address/model/session/FeeTest.java (hunk 3 above)
Comment (enhao25) [reply src/test/java/seedu/address/model/session/FeeTest.java]: Good suggestion! Have added the check and it is false as expected.

**Review by samleewy — APPROVED** [review]
looks good after the changes! 💯

File: docs/UserGuide.md [hunk 4]
```diff
@@ -122,8 +122,9 @@ The Home page is split into two views: **Reminder** and **Monthly Fees**.
 ##### Reminder
 The Reminder section displays tuition sessions within 3 days of the current date. Namely today, tomorrow and the day after tomorrow.
 
-##### Monthly Fees
-The Monthly Fees section displays tuition fees for the current and the past two months.
+##### 3 Months Monthly Fee
+The 3 Months Monthly Fees section displays tuition fees for the current month and the past two months.
+Note: The current month and past two months are calculated when the user first opens the application. No other months can be added to this section.
```
Comment (JonahhGohh) [inline docs/UserGuide.md:130]: For line 127, should we put this between a `<div markdown="block" class="alert alert-info">:information_source:</div>`?

**Review by JonahhGohh — APPROVED** [review]
LGTM, minor comment

File: docs/UserGuide.md (hunk 4 above)
Comment (enhao25) [reply docs/UserGuide.md:130]: Ahhhh yes! Good suggestion! Did that in the latest commit! Will push it up after it has completed the check!
